### PR TITLE
Replace `settings` variable with function

### LINKED ISSUE #130: Executor run CLI not respecting `--qc-compute-n-cores`
Running the [example](https://openff-bespokefit.readthedocs.io/en/latest/getting-started/quick-start.html#one-off-fits) in the docs using bespokefit version 0.1.0 installed via conda-forge sets up 2 qc workers but gives them both all available cores rather than the 8 specified via the CLI.

### PR DESCRIPTION
**Pull request by SimonBoothroyd**
## Description

This PR replaces the 'dangerous' `settings` field exposed by the bespoke executor with a more reliable function that returns the most up-to-date values.

The object returned by `current_settings()` should not be modified as the changes will have no effect. Instead the `apply_env` context manager should be used e.g.

```python
with Settings(
    BEFLOW_FRAGMENTER_WORKER_N_CORES=...,
    BEFLOW_FRAGMENTER_WORKER_MAX_MEM=...
).apply_env():

    settings = current_settings()
```

Possibly fixes #130 

## Status
- [x] Ready to go

### CODE REVIEW

File: openff/bespokefit/executor/services/qcgenerator/worker.py [hunk 1]
```diff
@@ -21,14 +21,16 @@
...
+
 redis_connection = redis.Redis(
-    host=settings.BEFLOW_REDIS_ADDRESS,
-    port=settings.BEFLOW_REDIS_PORT,
-    db=settings.BEFLOW_REDIS_DB,
+    host=__settings.BEFLOW_REDIS_ADDRESS,
+    port=__settings.BEFLOW_REDIS_PORT,
+    db=__settings.BEFLOW_REDIS_DB,
 )
 celery_app = configure_celery_app("qcgenerator", redis_connection)
```
Comment (jthorton) [inline openff/bespokefit/executor/services/qcgenerator/worker.py:35]: The info in the `redis_connection` object seems to be the same as that in the settings class and this logic seems to be repeated for each app, would it make sense to have the `configure_celery_app` just call the `current_settings` function and get the relavent info?

Comment (SimonBoothroyd) [reply openff/bespokefit/executor/services/qcgenerator/worker.py:35]: Yeah I think either having configure celery return a tuple of both the app and redis connection, or adding a new `configure_redis` method that also checks for the executor key you mentioned in slack / any other validation would be a great shout! 

I think I have a slight preference for the latter, but not by much 🙂

Comment (SimonBoothroyd) [reply openff/bespokefit/executor/services/qcgenerator/worker.py:35]: We should probably handle this in a new PR though so it doesn't get buried here! 😅

**Review by jthorton — APPROVED** [review]
@SimonBoothroyd thanks for the quick fix, can confirm this now works on linux as expected!

**Pull request merged**